### PR TITLE
Replace `serverspec` with `inspec` tests via. new CI matrix (inc. `semantic-release`)

### PR DESCRIPTION
**Pull request by myii**
Currently, there is no CI-based testing on this formula.  This PR brings in a number of the latest changes from the `template-formula`, in order to bring `inspec` testing via. a matrix of pre-salted images configured in Kitchen and run via. Travis.  In it's current state, 9/10 of the platforms pass the tests.  Here's a list of what remains before this can be merged:

* [x] Get remaining platform working (`ubuntu-1804`).
    * Didn't fix this but updated the matrix instead with the very latest available images to get 13/13 passing.
* [x] Update the `README` -- will do this once this PR is approved.
* [x] ~Recommends the merge of #256~.
    * Ignored.
* [x] ~Recommends the merge of #261~.
    * Merged.
* [x] Trim off the last commit once the pre-salted images are updated to provide `net-tools` by default.
* [x] Add new `centos-6` pre-salted image -- 14/14 working.
* [x] Test `use_upstream_repo`: `True` and `False`.
* [x] Test installing multiple PosgreSQL versions, including `9.6` and `10`.
* [x] Test `postgres.port`: `5432` and `5433`.
* [x] Add temporary workaround for `inspec` bug for `opensuse-leap-15` (`service`-based tests).4
* [x] Implement `semantic-release` (includes documentation and library for TOFS).

Note that the formula itself needs to be improved to get it working with `pillar.example` instead of the provided test pillar (if we want to go that way).  The improvements that need to be made can be seen by the commented out sections in the test pillar.

---

Update: Just a note about an issue with `serverspec`.  While testing out these ideas initially, tried to hook up the existing `serverspec` tests but ended up hitting a weird bug where it was trying to run `apt-get` on `opensuse` -- tracked it down to [this block](https://github.com/neillturner/kitchen-verifier-serverspec/blob/4146fe8131bddcac869d85f979d1ea1c511f8bba/lib/kitchen/verifier/serverspec.rb#L136-L145) in `kitchen-verifier-serverspec`.

### CODE REVIEW

**Comment by myii** [comment]
@daks Thanks for the review.  You're right about the commits -- I was planning to do that at the end.  The idea right now is that each commit is a part that may have changes depending on the review, so the PR can be modified as required.  For example, I definitely want to lose the last commit as well as maybe the one before that.  It's easier to make the necessary modifications while this is a fluid `WIP`.

I left the `semantic-release` commented out because it's definitely needed, whether now or very soon.  It opens up the discussion for it also.

Really appreciate you taking the time to review this PR.

**Comment by daks** [comment]
@vutny I think what it means is that during work (WIP PR) it's simpler to have one commit per change, it eases testing changes, reverting them if needed, etc. Then when work is finished, commits can be squashed.
I tend to use this pattern too.

**Comment by myii** [comment]
With 13/13 images passing and the `README` updated, this PR is now ready for final review.

**Comment by myii** [comment]
While 14/14 images being tested appears excessive, all sorts of combinations are there, besides different platforms and salt versions, as supplied by the pre-salted images:

* Testing with and without installing from the upstream repo.
* Testing multiple PostgreSQL versions.
* Testing multiple ports.

So if there is a requirement to remove some of these images, care must be exercised so worthwhile testing isn't lost.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+before_install:
+  - bundle install
+
+# Make sure the instances listed below match up with
+# the `platforms` defined in `kitchen.yml`
+env:
+  matrix:
+    - INSTANCE: default-debian-9-2019-2-py3
+    # - INSTANCE: default-ubuntu-1804-2019-2-py3
```
[the remark this answers is not in the document]
Comment (myii) [reply .travis.yml]: @daks This looks like it is going to take some effort to fix, so will probably need to leave this for a subsequent PR.

https://travis-ci.com/saltstack-formulas/postgres-formula/jobs/195597488#L1530:

```
       [INFO    ] Executing command ['systemd-run', '--scope', 'systemctl', 'start', 'postgresql.service'] in directory '/home/kitchen'
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

I mentioned in Slack:

> On the postgresql systemd issue, it appears that this has some relation to the service name being different on Ubuntu 18.04 onwards: `postgresql@10-main`.
> I.e. `postgresql@${version}-${cluster_name}`.

Logging in to the container, these commands actually work but hang there too.  So I'm not sure what extra steps are required to fix this.  Maybe @javierbertoli might have a suggestion.

**Comment by myii** [comment]
While not directly related to `serverspec` => `inspec`, adding `semantic-release` to this PR was very straightforward after all of the other changes made.  I've already tested it in my fork, the same way we have done with all of the other deployments.  This PR is ready to merge, depending on any final comments.

**Comment by myii** [comment]
Travis is hitting stuck builds on all three of the `opensuse` instances:

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

Tried restarting each of them to no avail.  There was nothing changed to have triggered this.

**Comment by myii** [comment]
Thanks to @javierbertoli, the Travis issue has been resolved and we're back to 13/13 images passing.  This is ready to merge again.

**Pull request merged**